modules/benchmarks/src/main/java/org/apache/ignite/internal/benchmarks/jmh/sql/JmhSqlBenchmark.java

```diff
@@ -0,0 +1,285 @@
+
+        client = Ignition.start(configuration("client").setClientMode(true));
+
+        cache = client.getOrCreateCache(new CacheConfiguration<Integer, Item>("CACHE")
+            .setIndexedTypes(Integer.class, Item.class)
+            .setAffinity(new RendezvousAffinityFunction(false, PARTS_CNT))
+        );
+
+        try (IgniteDataStreamer<Integer, Item> ds = client.dataStreamer("CACHE")) {
+            for (int i = 0; i < KEYS_CNT; i++)
+                ds.addData(i, new Item(i));
+        }
+    }
+
+    /**
+     * Stop Ignite instance.
+     */
+    @TearDown
+    public void tearDown() {
+        client.close();
+
+        for (Ignite ignite : servers)
+            ignite.close();
+    }
+
+    /**
+     * Query unique value (full scan).
+     */
+    @Benchmark
+    public void querySimpleUnique() {
+        int key = ThreadLocalRandom.current().nextInt(KEYS_CNT);
+
+        List<?> res = executeSql("SELECT name FROM Item WHERE fld=?", key);
+
+        assert res.size() == 1;
+    }
+
+    /**
+     * Query unique value (indexed).
+     */
+    @Benchmark
+    public void querySimpleUniqueIndexed() {
+        int key = ThreadLocalRandom.current().nextInt(KEYS_CNT);
+
+        List<?> res = executeSql("SELECT name FROM Item WHERE fldIdx=?", key);
+
+        assert res.size() == 1;
+    }
+
+    /**
+     * Query batch (full scan).
+     */
+    @Benchmark
+    public void querySimpleBatch() {
+        int key = ThreadLocalRandom.current().nextInt(KEYS_CNT);
+
+        List<?> res = executeSql("SELECT name FROM Item WHERE fldBatch=?", key / BATCH_SIZE);
+
+        assert res.size() == BATCH_SIZE;
+    }
+
+    /**
+     * Query batch (indexed).
+     */
+    @Benchmark
+    public void querySimpleBatchIndexed() {
+        int key = ThreadLocalRandom.current().nextInt(KEYS_CNT);
+
+        List<?> res = executeSql("SELECT name FROM Item WHERE fldIdxBatch=?", key / BATCH_SIZE);
+
+        assert res.size() == BATCH_SIZE;
+    }
+
+    /**
+     * Query with group by and aggregate.
+     */
+    @Benchmark
+    public void queryGroupBy() {
+        List<?> res = executeSql("SELECT fldBatch, AVG(fld) FROM Item GROUP BY fldBatch");
+
+        assert res.size() == KEYS_CNT / BATCH_SIZE;
+    }
+
+    /**
+     * Query with indexed field group by and aggregate.
+     */
+    @Benchmark
+    public void queryGroupByIndexed() {
+        List<?> res = executeSql("SELECT fldIdxBatch, AVG(fld) FROM Item GROUP BY fldIdxBatch");
+
+        assert res.size() == KEYS_CNT / BATCH_SIZE;
+    }
+
+    /**
+     * Query with sorting (full set).
+     */
+    @Benchmark
+    public void queryOrderByFull() {
+        List<?> res = executeSql("SELECT name, fld FROM Item ORDER BY fld DESC");
+
+        assert res.size() == KEYS_CNT;
+    }
+
+    /**
+     * Query with sorting (batch).
+     */
+    @Benchmark
+    public void queryOrderByBatch() {
+        int key = ThreadLocalRandom.current().nextInt(KEYS_CNT);
```

Comment:
Instead of this rnd implementstion it`s more clear to use pre generated arr of pseudo random algo, such already exist in some of utility class but i can`t found it ) Count of operations must be predicted.

Reply:
The count of operations is the same in this case, no matter what key we are using, the index scan will traverse about the same amount of nodes and produce the same amount of rows.

Reply:
we have **for loop** in **ThreadLocalRandom** implementation, i talk about escaping of it, but it just a minor of course, let store as is.